pkg/storaged/crypto-keyslots.jsx

```diff
@@ -312,15 +493,24 @@
         ].concat(existing_passphrase_fields(_("Saving a new passphrase requires unlocking the disk. Please provide a current disk passphrase."))),
         Action: {
             Title: _("Add"),
-            action: function (vals) {
+            action: function (vals, progress) {
                 const existing_passphrase = vals.passphrase || recovered_passphrase;
                 if (!client.features.clevis || vals.type == "luks-passphrase") {
                     return passphrase_add(block, vals.new_passphrase, existing_passphrase);
                 } else {
-                    return get_tang_adv(vals.tang_url).then(function (adv) {
-                        edit_tang_adv(client, block, null,
-                                      vals.tang_url, adv, existing_passphrase);
-                    });
+                    return get_tang_adv(vals.tang_url)
+                            .then(adv => {
+                                const steps = [];
+                                return ensure_nbde_support(steps, progress, client, block)
+                                        .then(() => {
+                                            if (steps.length > 0)
+                                                ensure_nbde_support_dialog(steps, client, block, vals.tang_url,
+                                                                           adv, existing_passphrase);
+                                            else
+                                                edit_tang_adv(client, block, null,
+                                                              vals.tang_url, adv, existing_passphrase);
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17796-20221020-065227-542cbe2b-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)